File: shared/src/main/java/com/gcgenome/lims/data/Analysis.java
package com.gcgenome.lims.data;

import jsinterop.annotations.*;
import jsinterop.base.Js;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@JsType(isNative=true, namespace= JsPackage.GLOBAL, name="Object")
@Setter(onMethod_={@JsOverlay, @JsIgnore})
@Getter(onMethod_={@JsOverlay, @JsIgnore})
@Accessors(fluent=true)
public final class Analysis {
	private String 			batch;
	private Double 			row;
	private Request 		request;
	private String 			sample;
	private Double 			freemix;
	@JsProperty(name="raw_reads_millions")
	private Double 			rawReadsMillions;
	@JsProperty(name="dup_rate")
	private Double 			dupRate;
	private Double 			gc;
	@JsProperty(name="total_reads")
	private Double 			totalReads;
	private Double 			mean;
	private Double 			median;
	private String 			qc;
	@JsProperty(name="chr_xcnt")
	private Double 			chrXCnt;
	@JsProperty(name="chr_ycnt")
	private Double 			chrYCnt;
	@JsProperty(name="chr_xprop")
	private Double 			chrXProp;
	@JsProperty(name="chr_yprop")
	private Double 			chrYProp;
	@JsProperty(name="pred_sex")
	private String			predSex;
	@JsProperty(name="freemix_tmp")
	private Double 			freemixTmp;
	@JsProperty(name="raw_reads_millions_tmp")
	private Double 			rawReadsMillionsTmp;
	@JsProperty(name="dup_rate_tmp")
	private Double 			dupRateTmp;
	@JsProperty(name="gc_tmp")
	private Double 			gcTmp;
	@JsProperty(name="total_reads_tmp")
	private Double 			totalReadsTmp;
	@JsProperty(name="mean_tmp")
	private Double 			meanTmp;
	@JsProperty(name="median_tmp")
	private Double 			medianTmp;
	@JsProperty(name="qc_tmp")
	private String 			qcTmp;
	@JsProperty(name="chr_xcnt_tmp")
	private Double 			chrXCntTmp;
	@JsProperty(name="chr_ycnt_tmp")
	private Double 			chrYCntTmp;
	@JsProperty(name="chr_xprop_tmp")
	private Double 			chrXPropTmp;
	@JsProperty(name="chr_yprop_tmp")
	private Double 			chrYPropTmp;
	@JsProperty(name="pred_sex_tmp")
	private String			predSexTmp;
	@JsProperty(name="too5_pred")
	private String 			too5Pred;
	@JsProperty(name="too5_fems_prob")
	private Double 			too5FemsProb;
	@JsProperty(name="too6_pred")
	private String 			too6Pred;
	@JsProperty(name="too6_fems_prob")
	private Double 			too6FemsProb;
	private Double 			iscore;
	@JsProperty(name="cad_ensemble_prob")
	private Double 			cadEnsembleProb;
	private String 			result;
	@JsProperty(name="created_at")
	private String 			createAt;
	@JsProperty(name="last_modify_at")
	private String 			lastModifyAt;
	private User 			lastModifyBy;
	private String 			etc;
	private Report 			report;
	@JsOverlay
	@JsIgnore
	public Integer row() {
		if(row == null) return null;
		else return row.intValue();
	}
	@JsOverlay
	@JsIgnore
	public Request request() {
		if(request == null) return null;
		return Js.uncheckedCast(request);
	}
}

File: analysis/src/main/kotlin/com/gcgenome/lims/data/Analysis.kt
package com.gcgenome.lims.data

data class Analysis(
    val sample:                 Long,
    val service:                String,
    val batch:                  String,
    val row:                    Int,
) {
    var request:                Request? = null
    var report:                 Report? = null
    var file:                   String? = ""
    var createdAt:              String? = ""
    var lastModifyAt:           String? = ""
    var lastModifyBy:           User? = null

    var freemix:                Double = 0.0
    var rawReadsMillions:       Long =  0
    var dupRate:                Double = 0.0
    var gc:                     Double = 0.0
    var totalReads:             Double = 0.0
    var mean:                   Double = 0.0
    var median:                 Double = 0.0
    var qc:                     String = "F"
    var chrXCnt:                Long = 0
    var chrYCnt:                Long = 0
    var chrXProp:               Double = 0.0
    var chrYProp:               Double = 0.0
    var predSex:                String = "?"

    var freemixTmp:             Double = 0.0
    var rawReadsMillionsTmp:    Long =  0
    var dupRateTmp:             Double = 0.0
    var gcTmp:                  Double = 0.0
    var totalReadsTmp:          Double = 0.0
    var meanTmp:                Double = 0.0
    var medianTmp:              Double = 0.0
    var qcTmp:                  String = "F"
    var chrXCntTmp:             Long = 0
    var chrYCntTmp:             Long = 0
    var chrXPropTmp:            Double = 0.0
    var chrYPropTmp:            Double = 0.0
    var predSexTmp:             String = "?"

    var too5Pred:               String = ""
    var too5FemsProb:           Double = 0.0
    var too6Pred:               String = ""
    var too6FemsProb:           Double = 0.0
    var iscore:                 Double = 0.0
    var cadEnsembleProb:        Double = 0.0
    var result:                 String = ""
    var comment:                String = ""
    var femsCovBc:              Double = 0.0
    var femsBc:                 Double = 0.0
    var covBc:                  Double = 0.0
    var femsCovBernn:           Double = 0.0
    var femsPath:               String = ""
    var iscorePath:             String = ""
    var language:               String = ""
    var clinicalCancer:         String = ""
    var cfDnaConcentration:     Double = 0.0
}
